File: ts_to_heifer/test/heifer_output/01_basic_types.ml
=== Translation Successful ===

Heifer IR:
let identity_number = fun x (*@ req x:int; ens res:int @*) -> x in
let identity_string = fun s (*@ req s:Str; ens res:Str @*) -> s in
let identity_bool = fun b (*@ req b:bool; ens res:bool @*) -> b in
let log_message = fun msg (*@ req msg:Str; ens res:() @*) -> log msg in
let add = fun x y (*@ req x:int/\y:int; ens res:int @*) -> (x + y) in
let format = fun prefix value (*@ req prefix:Str/\value:int; ens res:Str @*) -> let tmp1 = toString  in
(prefix + tmp1) in
()

Type: unit

✓ Output is in Heifer's native format

File: ts_to_heifer/test/test_cases/01_basic_types.ts
// Test 1: Basic primitive types
// Testing how TypeScript parser handles basic type annotations

// Simple number parameter and return
function identity_number(x: number): number {
  return x;
}

// String types
function identity_string(s: string): string {
  return s;
}

// Boolean types
function identity_bool(b: boolean): boolean {
  return b;
}

// Void return type
function log_message(msg: string): void {
  console.log(msg);
}

// Multiple parameters
function add(x: number, y: number): number {
  return x + y;
}

// Multiple parameters with different types
function format(prefix: string, value: number): string {
  return prefix + value.toString();
}
